File: core/app_identity.py
from __future__ import annotations

from pathlib import Path


STANDARD_APP_NAME = "Whisper"
STANDARD_BUNDLE_ID = "dev.whisper.local"
STANDARD_STATE_DIR = ".whisper"
STANDARD_KEYCHAIN_SERVICE = "com.whisper.api-keys"

def app_name() -> str:
    return STANDARD_APP_NAME


def bundle_id() -> str:
    return STANDARD_BUNDLE_ID


def state_dir() -> Path:
    return Path.home() / STANDARD_STATE_DIR


def keychain_service() -> str:
    return STANDARD_KEYCHAIN_SERVICE

File: swift/TalkType/TalkType/AppIdentity.swift
import Foundation

enum AppIdentity {
    static let appName = "TalkType"
    static let bundleID = "dev.talktype.local"

    static let legacyAppName = "Whisper"
    static let legacyStateDir = ".whisper"
    static let standardStateDir = ".talktype"

    static let keychainService = "com.talktype.api-keys"
    static let legacyKeychainService = "com.whisper.api-keys"

    static var stateDir: URL {
        let home = FileManager.default.homeDirectoryForCurrentUser
        let target = home.appendingPathComponent(standardStateDir)
        let legacy = home.appendingPathComponent(legacyStateDir)

        if FileManager.default.fileExists(atPath: target.path) {
            return target
        }
        if FileManager.default.fileExists(atPath: legacy.path) {
            do {
                try FileManager.default.moveItem(at: legacy, to: target)
                return target
            } catch {
                return legacy
            }
        }
        return target
    }
}
